braidpool-primitives/src/utils/bitcoin.rs

```diff
@@ -10,23 +10,44 @@ type MerkleRoot = TxMerkleNode;
 
 pub struct MerklePathProof {
     pub transaction_hash: Txid,
+    pub is_right_leaf: bool,
     pub merkle_path: Vec<TxMerkleNode>,
```

Comment:
```suggestion
    pub is_right_leaf: bool,
    pub is_alone_leaf: bool,
    pub merkle_path: Vec<TxMerkleNode>,
```

I'm not sure if that will ever happen for `coinbase` or `payout` but we still need to handle one more case.
When there's an odd number of leaves, we need to handle the case for the last leaf where we do `HASH256(A + A)`

Reply:
Yeppp, but this would be already handled because the merkle path, will consist of that copy of the transaction, since the binary tree is always complete.

Reply:
Okk! Fine.

Reply:
perfect :)